### PR TITLE
Hygiene: Fix or disable all warnings, enable Warnings-As-Errors

### PR DESCRIPTION
**Pull request by lesaltzm**

### CODE REVIEW

File: src/tests/Microsoft.PowerFx.Interpreter.Tests/RecalcEngineAsyncTests.cs [hunk 1]
```diff
@@ -19,14 +19,16 @@ namespace Microsoft.PowerFx.Tests
...
     {
-        // Trivial async function that runs synchronously. 
+        // Intentionally trivial async function that runs synchronously. 
+#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
         private static async Task<FormulaValue> Worker(FormulaValue[] args, CancellationToken cancel)
         {
             var n = (NumberValue)args[0];
 
             var result = FormulaValue.New(n.Value * 2);
             return result;
```
Comment (CarlosFigueiraMSFT) [inline src/tests/Microsoft.PowerFx.Interpreter.Tests/RecalcEngineAsyncTests.cs:29]: >result [](http://example.com/codeflow?start=19&length=6)

You can use `return Task.FromResult(result);` - and that would remove the need to use the `async` method modifier

**Review by CarlosFigueiraMSFT — DISMISSED** [review]
:shipit:

File: src/tests/Microsoft.PowerFx.Interpreter.Tests/RecalcEngineAsyncTests.cs [hunk 2]
```diff
@@ -19,14 +19,16 @@ namespace Microsoft.PowerFx.Tests
     // Test async eval features. 
     public class RecalcEngineAsyncTests : PowerFxTest
     {
-        // Trivial async function that runs synchronously. 
+        // Intentionally trivial async function that runs synchronously. 
+#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
```
Comment (MikeStall) [inline src/tests/Microsoft.PowerFx.Interpreter.Tests/RecalcEngineAsyncTests.cs:23]: Can we remove this warning?
async keyword actually matters for how AggregateException is handled.

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestHelpers/TestRunner.cs [hunk 3]
```diff
@@ -242,7 +242,7 @@ public TestRunFullResults RunTests()
                 {
                     var engineName = runner.GetName();
 
-                    var (result, msg) = runner.RunAsync(testCase).Result;
```
Comment (MikeStall) [inline src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestHelpers/TestRunner.cs:245]: >Result [](http://example.com/codeflow?start=66&length=6)

double check?

Comment (lesaltzm) [reply src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestHelpers/TestRunner.cs:245]: Intended, runasync isn't actually async (I'll fix the name)

**Review by MikeStall — CHANGES_REQUESTED** [review]
:clock1:

File: src/libraries/Microsoft.PowerFx.Interpreter/EvalVisitor.cs [hunk 4]
```diff
@@ -316,14 +316,14 @@ private ValueTask<FormulaValue> VisitBinaryOpNode(BinaryOpNode node, SymbolConte
...
-                    return OperatorDynamicGetField(node, args);
+                    return new ValueTask<FormulaValue>(OperatorDynamicGetField(node, args));
 
                 default:
                     return new ValueTask<FormulaValue>(CommonErrors.UnreachableCodeError(node.IRContext));
             }
         }
 
-        private static async ValueTask<FormulaValue> OperatorDynamicGetField(BinaryOpNode node, FormulaValue[] args)
+        private static FormulaValue OperatorDynamicGetField(BinaryOpNode node, FormulaValue[] args)
```
Comment (CarlosFigueiraMSFT) [inline src/libraries/Microsoft.PowerFx.Interpreter/EvalVisitor.cs:326]: >OperatorDynamicGetField [](http://example.com/codeflow?start=36&length=23)

Not related to your changes, but all other operators are async functions (using the error wrapper) - we should make this one as well (and simplify the blank/error handling on it. Can be dealt with in a separate PR

**Review by CarlosFigueiraMSFT — APPROVED** [review]
:shipit:

**Pull request merged**